### PR TITLE
Add stackset and stack metrics

### LINKED ISSUE #146: Report stackset/stack information via Prometheus metrics
We should report per-stack (and maybe per-stackset) metrics for various settings, similar to how Kubernetes itself behaves. Reconstructing this from logs and events is very annoying. Things that we could start with:
 - desired traffic weight
 - actual traffic weight
 - prescaling active (as a gauge; 0/1)
 - prescaling replicas

### PR DESCRIPTION
**Pull request by aermakov-zalando**
Adds the following metrics to help with debugging:
 * `stackset_stackset_stacks` (stack count)
 * `stackset_stack_desired_traffic_weight`, `stackset_stack_actual_traffic_weight`
 * `stackset_stack_ready` (0.0/1.0)
 * `stackset_stack_prescaling_active` (0.0/1.0)
 * `stackset_stack_prescaling_replicas`

Fix #146.

### CODE REVIEW

File: pkg/core/metrics.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+		}
+	}
+	return nil
+}
+
+func extractLabels(nameKey string, obj metav1.Object) prometheus.Labels {
+	return prometheus.Labels{
+		"namespace":   obj.GetNamespace(),
+		nameKey:       obj.GetName(),
+		"application": obj.GetLabels()["application"],
```
Comment (arjunrn) [inline pkg/core/metrics.go]: Won't this `panic` when `application` label is missing?

Comment (aermakov-zalando) [reply pkg/core/metrics.go]: No, it's go. It'll just return an empty string.

**Comment by arjunrn** [comment]
👍

**Comment by aermakov-zalando** [comment]
👍

**Pull request merged**